core/metacling/src/TCling.cxx

```diff
@@ -1447,39 +1447,39 @@ static std::string FindLibraryName(void (*func)())
    {
       return {};
    }
-   return ROOT::TMetaUtils::ResolveSymlink(moduleName);
+   return ROOT::TMetaUtils::GetRealPath(moduleName);
```

Comment:
This function is a bit too complex with all these places for returning with a function call. This is probably one of the few cases where setting the name to be returned and using a `goto out;` where you call `return ROOT::TMetaUtils::GetRealPath(name);` just once is a good idea.